### PR TITLE
Moved indexedAt time to record instead of ipld-block

### PR DESCRIPTION
**Pull request by dholms**
This migration moves the `indexedAt` time on `ipld_block` to the relevant `record`

This makes the migration in https://github.com/bluesky-social/atproto/pull/500 much simpler (no need to track indexedAt time) & is also fixes a leaky abstraction - `ipld_block` is a repo storage concept & `record` is an indexing layer concept.

### CODE REVIEW

File: packages/pds/tests/migrations/indexed-at-on-record.test.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+describe('indexedAt on record migration', () => {
+  let db: Database
+  let rawDb: Kysely<any>
+
+  beforeAll(async () => {
+    if (process.env.DB_POSTGRES_URL) {
+      db = Database.postgres({
+        url: process.env.DB_POSTGRES_URL,
+        schema: 'migration_duplicate_records',
```
Comment (devinivy) [inline packages/pds/tests/migrations/indexed-at-on-record.test.ts:15]:
```suggestion
        schema: 'migration_indexed_at_on_record',
```

**Review by devinivy — APPROVED** [review]
Lookin good!

File: packages/pds/tests/migrations/indexed-at-on-record.test.ts (hunk 1 above)
Comment (dholms) [reply packages/pds/tests/migrations/indexed-at-on-record.test.ts:15]: ugh woops 🤦

File: packages/pds/src/db/migrations/20230127T215753149Z-indexed-at-on-record.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  const ref = db.dynamic.ref
+
+  await db
+    .updateTable('record')
+    .set({
+      indexedAt: db
+        .selectFrom('ipld_block')
+        .whereRef('ipld_block.cid', '=', ref('record.cid'))
+        .select('indexedAt'),
```
[the remark this answers is not in the document]
Comment (dholms) [reply packages/pds/src/db/migrations/20230127T215753149Z-indexed-at-on-record.ts:18]: ah nice. great point 👌

**Pull request merged**